> > When the host parameter is not filled in, the config. get method will throw a null pointer exception
> 
> Could you provide the error stack?

![image](https://github.com/user-attachments/assets/278e7ff2-afd6-4166-9c45-67ac536e590b)

![image](https://github.com/user-attachments/assets/2162cae4-b286-45fd-8a58-a14e59d54c51)






**stack info**
Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
	at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:228)
	at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
	at org.apache.seatunnel.example.engine.SeaTunnelEngineLocalExample.main(SeaTunnelEngineLocalExample.java:52)
Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a source for identifier 'Redis'.
	at org.apache.seatunnel.api.table.factory.FactoryUtil.restoreAndPrepareSource(FactoryUtil.java:168)
	at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:86)
	at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:376)
	at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:227)
	at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:123)
	at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:191)
	at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:165)
	... 2 more
Caused by: java.lang.NullPointerException
	at org.apache.seatunnel.connectors.seatunnel.redis.config.RedisParameters.buildWithConfig(RedisParameters.java:72)
	at org.apache.seatunnel.connectors.seatunnel.redis.source.RedisSource.<init>(RedisSource.java:56)
	at org.apache.seatunnel.connectors.seatunnel.redis.source.RedisSourceFactory.lambda$createSource$0(RedisSourceFactory.java:45)
	at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:180)
	at org.apache.seatunnel.api.table.factory.FactoryUtil.restoreAndPrepareSource(FactoryUtil.java:136)
	... 8 more
[] 2025-03-03 18:10:02,703 DEBUG org.apache.hadoop.util.ShutdownHookManager - ShutdownHookManger completed shutdown.
Disconnected from the target VM, address: '127.0.0.1:55053', transport: 'socket'

Process finished with exit code 1

--- a/seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java
+++ b/seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/config/RedisParameters.java
@@ -46,7 +46,7 @@ import static org.apache.seatunnel.connectors.seatunnel.redis.exception.RedisErr
 @Slf4j
 public class RedisParameters implements Serializable {
     private String host;
-    private int port;
+    private Integer port;
     private String auth = "";
     private int dbNum;
     private String user = "";
@@ -68,10 +68,8 @@ public class RedisParameters implements Serializable {
     public void buildWithConfig(ReadonlyConfig config) {
         // set host
         this.host = config.get(RedisBaseOptions.HOST);
-        if (config.getOptional(RedisBaseOptions.PORT).isPresent()) {
-            // set port
-            this.port = config.get(RedisBaseOptions.PORT);
-        }
+        // set port
+        this.port = config.get(RedisBaseOptions.PORT);
         // set db_num
         this.dbNum = config.get(RedisBaseOptions.DB_NUM);
         // set hash key mode

